docs/UserGuide.md

```diff
@@ -2,133 +2,411 @@
+
+    add “DEADLINE_NAME” <DATE> <TIME>
+
+* Dealine name should be in a pair of quotation marks.
+
+* Either `<DATE>` or `<TIME>` or both of them are required here.
+
+* Formats of `DATE` and `TIME` are the same as what we use to create event-like
+  tasks.
+
+* The default value of `DATE` will be the same as the real date on user's
+  computer.
+
+* There is no default value for `TIME` as long as the `DATE` passed in is
+  valid.
+
+#### Examples
+
+* `add "CS2103 V0.1" 6/10`
+
+* `add "event proposal" 1800`
+
+* `add "EE2024 homework 1" 1/11/2016 600`
+
+### Deleting a floating task/event/deadline: `del`
+
+Delete a useless floating task/event/deadline on TaskTracker.
+
+    del "TASK_NAME"
+
+* Task name should be in a pair of quotation marks.
+
+* Here we assume that no duplicate task name is used among all the task.
+
+#### Examples
+
+* `del "project meeting"`
+
+* `del "CS2102 week 8 lecture"`
+
+* `del "reading a book"`
+
+### Edit a floating task: `edit`
+
+Edit a floating task to revise its name or priority.
+
+    edit "TASK_NAME" [n-NEW_NAME | p-PRIORITY]...
+
+* Task name should be in a pair of quotation marks.
+
+* `PRIORITY` should only be the interger ranges from `0` to `5`.
+
+* Here we assume that no duplicate task name is used among all the task.
+
+#### Examples
+
+* `edit "washing clothes" p-0`
+
+* `edit "buy textbooks" n-buy stationary`
+
+* `edit "go shopping" n-"go to Nanyang Mart" p-1`
+
+### Edit an event : `edit`
+
+Edit an event to revise its name, starting/ending date/time and location.
+
+    edit "TASK_NAME" [sd-NEW_START_TIME | st-NEW_START_DATE | ed-NEW_END_DATE | et-NEW_END_TIME | n-NEW_NAME | loc-NEW_LOCATION]...
+
+* Task name should be in a pair of quotation marks.
+
+* `[sd-NEW_START_TIME | st-NEW_START_DATE | ed-NEW_END_DATE | et-NEW_END_TIME]`
+  are of the same format when creating event-like task.
+
+#### Examples
+
+* `edit "Excel Workshop" loc-LT6`
+
+* `edit "3D printing training" st-1600 et-1800`
+
+* `edit "project discussion" n-proposal meeting st-1900`
+
+### Edit a dealine : `edit`
+
+Edit a deadline to revise its name and due date/time.
+
+    edit "TASK_NAME" [dd-DUE_DATE | dt-DUE_TIME | n-NEW_NAME]...
+
+* Task name should be in a pair of quotation marks.
+
+* `[dd-DUE_DATE | dt-DUE_TIME]` are of the same formate when creating
+  event-like task.
+
+#### Examples
+
+* `edit "essay assignment 1" dt-1700`
+
+* `edit "write email to prof." dd-23/11/2016`
+
+### Mark a floating task/deadline as done/finished: `fin`
+
+Mark a floating task/event/deadline as done on TaskTracker, the marked tasks
+will be archived.
+
+    fin "TASK_NAME"
+
+* Task name should be in a pair of quotation marks.
+
+* Deadlines that have already passed it `DUE_TIME` will be marked as done
+  automatically.
+
+#### Examples
+
+* `fin "project meeting"`
+
+* `fin "watch a movie"`
+
+### Show empty time slots : `slot`
+
+Show all empty time slots in a given time period with a given duration.
```

Comment:
This would be for the PowerSearch feature? The search keywords function seems very basic though... note that it's better to have a few good features than have more half-baked features so make sure any features you add are well-thought-out